File: brian_ortega_delahoz/Clases/Programación/Junio2024/7-Jun-24/recaudo.dart
import 'dart:io';

void main() {
 
  int cantotal = 0;
  double prom = 0;
  int estudiantesmas5 = 0;
  int cantidadmas5 = 0;
  double promMas5 = 0;
  int estudiantesmenos5 = 0;
  int cantidadmenos5 = 0;
  double promMenos5 = 0;
  
  print('Números de aprendices');
  int estudiantes = int.parse(stdin.readLineSync()!);

  for (int i = 1; i <= estudiantes; i++) {
   
    print('Ingrese el aporte hecho por el aprendiz #$i:');
    int aporte = int.parse(stdin.readLineSync()!);
    cantotal += aporte;
    prom = (cantotal / estudiantes);

   
    if (aporte >= 5000) {
      estudiantesmas5++;
      cantidadmas5 += aporte;
      promMas5 = cantidadmas5 / estudiantesmas5;

    } else {
      estudiantesmenos5++;
      cantidadmenos5 += aporte;
      promMenos5 = cantidadmenos5 / estudiantesmenos5;
    }
  }

  print('La cantidad total recaudada fue de $cantotal \n');
  print('El promedio general es: $prom \n');
  print(
      'La cantidad de aprendices que donaron 5000 o más, fueron $estudiantesmas5 y con estos estudiantes fue recaudado: $cantidadmas5 con un promedio de $promMas5 \n');
  print(
      'La cantidad de aprendices que donaron menos de 5000, fueron $estudiantesmenos5 y con estos estudiantes fue recaudado: $cantidadmenos5 con un promedio de $promMenos5');
}

File: brian_ortega_delahoz/Clases/Programación/Junio2024/7-Jun-24/recaudo.rb

contadormas5000 = 0
acummas5000 = 0

contadormenos5000 = 0
acummenos5000 = 0

puts "Ingresa el número de Aprendices"
num = gets.chomp.to_i

for i in 1..num
    puts "Ingrese el aporte hecho por el aprendiz #$i:"
    aporte = gets.chomp.to_f
    
    if aporte >= 5000
        contadormas5000 = contadormas5000 + 1
        acummas5000 = acummas5000 + aporte
    else
        contadormenos5000 = contadormenos5000 + 1
        acummenos5000 = acummenos5000 + aporte
    end
end

total = acummas5000 +acummenos5000
prom = total / num
prom2 = acummas5000 / contadormas5000
prom3 = acummenos5000 / contadormenos5000


puts "-La cantidad total recogida es: #{total}."
puts "-El promedio del total recogido es: #{prom}"
puts "-Personas que aportaron 5.000 o más: #{contadormas5000}."
puts "-Cantidad total de los que aportaron 5.000 o más: #{acummas5000}."
puts "-Promedio de aporte de los que dieron 5.000 o más: #{prom2}."
puts "-Personas que aportaron menos de 5.000: #{contadormenos5000}."
puts "-Cantidad total de los que aportaron menos de 5.000: #{acummenos5000}."
puts "-Promedio de aporte de los que dieron menos de 5.000: #{prom3}."
